src/metrics/peer.go

```diff
@@ -109,6 +115,15 @@ func (pm *Peer) GetAllMessagesCount() uint64 {
 		pm.ProposerSlashing.Cnt)
 }
 
+// Update beacon Status of the peer
+func (pm *Peer) UpdateBeaconStatus(bStatus beacon.Status) {
```

Comment:
As I can see from [here](https://github.com/protolambda/zrnt/blob/9ef829e4a73f094061677cc6846f3dcdedb7a439/eth2/beacon/common/network.go#L235) you are storing the whole `beacon.Status`. Why not just:

```go
pm.BeaconStatus = bStatus
```

Reply:
Yep, much easier, I would like to add some middle struct with the time-stamp, so that I can implement this version tracker.